src/check.c

```diff
@@ -149,10 +150,113 @@ TCase *tcase_create(const char *name)
     tc->ch_sflst = check_list_create();
     tc->unch_tflst = check_list_create();
     tc->ch_tflst = check_list_create();
+    tc->tags = NULL;
+    tc->num_tags = 0;
 
     return tc;
 }
 
+void tcase_set_tags(TCase * tc, const char *tags_orig)
+{
+    unsigned int count = 0;
+    char *tags;
+    char *tag;
+
+    /* free any existing tags */
+    if (NULL != tc->tags)
+    {
+	unsigned int i;
+
+	for (i = 0; i < tc->num_tags; i++)
+	{
+	    free(tc->tags[i]);
+	} 
+	free(tc->tags);
+	tc->tags = NULL;
+    }
+    tc->num_tags = 0;
+
+    if (NULL == tags_orig)
+    {
+	return;
+    }
+
+    /* first pass to count the number of tags to be added */
+    tags = strdup(tags_orig);
+    tag = strtok(tags, " ");
+    while (tag)
+    {
+	if (strlen(tag))
+	{
+	    count++;
+	}
+	tag = strtok(NULL, " ");
+    }
+    free(tags);
+
+    if (0 == count)
+    {
+	return;
+    }
+    
+    tc->tags = calloc(count, sizeof(const char *));
```

Comment:
You will need to do a cast here to avoid an error from clang:
```
 check.c:202:14: fatal error: assigning to 'char **' from incompatible type 'void *'; 
    tc->tags = calloc(count, sizeof(const char *));
```